src/cmd/linuxkit/pkglib/build.go

```diff
@@ -191,3 +201,62 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 
 	return nil
 }
+
+type buildCtx struct {
+	sources []string
+}
+
+// Copy iterates over the sources, tars up the content after rewriting the paths.
+// It assumes that sources is sane, ie is well formed and the first part is an absolute path
+// and that it exists. NewFromCLI() ensures that.
+func (c *buildCtx) Copy(w io.WriteCloser) {
+	tw := tar.NewWriter(w)
+	defer func() {
+		tw.Close()
+		w.Close()
+	}()
+
+	for _, source := range c.sources {
+		tmp := strings.Split(source, ":")
+		src := tmp[0]
+		dst := tmp[1]
+		log.Debugf("Adding to build context: %s -> %s", src, dst)
+
+		f := func(p string, i os.FileInfo, err error) error {
+			if err != nil {
+				log.Warnf("ctx: Walk error on %s: %v", p, err)
+				return nil
+			}
+
+			if i.IsDir() {
+				return nil
+			}
+
+			f, err := os.Open(p)
+			if err != nil {
+				log.Warnf("ctx: Open %s: %v", p, err)
+				return nil
+			}
+			defer f.Close()
+
+			h, err := tar.FileInfoHeader(i, "")
```

Comment:
In `linuxkit build` we force `PAX` headers, IIRC for reasons to do with path length limitations. Should we do the same here?

Reply:
turns out `dockerd` doesn't like it:
```
Sending build context to Docker daemon   16.9kB
Error response from daemon: Dockerfile parse error line 1: unknown instruction: /PAXHEADERS.00000000000000000000000000000013200000000000010217
exit status 1
```

Reply:
Interesting!

I guess the tarfile is all relative to the root of the build context, so it won't matter if `cwd` is super long and we only worry about very deep paths within the actual context itself (which is less likely).